Add CI Visibility custom metrics section to dotnet docs. (#18993)
DataDog/documentation: content/en/continuous_integration/tests/dotnet.md

### Adding custom tags to tests

> <div class="alert alert-warning">
>   <strong>Note:</strong> To add custom tags to tests, you must configure <a href="#custom-instrumentation">Custom Instrumentation</a> first.
> </div>

Reviewer: ```suggestion
To add custom tags to tests, configure [custom instrumentation](#custom-instrumentation) first.
```

To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][7] section of the .NET custom instrumentation documentation.

### Adding custom metrics to tests

<div class="alert alert-warning">
  <strong>Note:</strong> To add custom metrics to tests, you must configure <a href="#custom-instrumentation">Custom Instrumentation</a> first.
</div>

Just like tags, you can add custom metrics to your tests by using the current active span:

```csharp
// inside your test
var scope = Tracer.Instance.ActiveScope; // from Datadog.Trace;
if (scope != null) {
    scope.Span.SetTag("memory_allocations", 16);
}
// test continues normally
// ...
```

> To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][7] section of the .NET custom instrumentation documentation.

Reviewer 2: ```suggestion
To create filters or visualizations for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][7] section of the .NET custom instrumentation documentation.
```
I don't think you can group by a measure 

Also, here we should link to the guide here https://github.com/DataDog/documentation/pull/18831

Reviewer 2: https://github.com/DataDog/documentation/pull/18831 is merged, so you can link to the guide

To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][7] section of the .NET custom instrumentation documentation.

### Adding custom metrics to tests

> <div class="alert alert-warning">
>   <strong>Note:</strong> To add custom metrics to tests, you must configure <a href="#custom-instrumentation">Custom Instrumentation</a> first.
> </div>

Reviewer: ```suggestion
To add custom metrics to tests, configure [custom instrumentation](#custom-instrumentation) first.
```

Just like tags, you can add custom metrics to your tests by using the current active span:

```csharp
// inside your test
var scope = Tracer.Instance.ActiveScope; // from Datadog.Trace;
if (scope != null) {
    scope.Span.SetTag("memory_allocations", 16);
}
// test continues normally
// ...
```

To create filters or visualizations for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][7] section of the .NET custom instrumentation documentation.

> Read more about custom metrics in [Add Custom Metrics Guide][15].

Reviewer: ```suggestion
Read more about custom metrics in the [Add Custom Metrics Guide][15].
```
